packages/shared/src/write-archive/index.ts

```diff
@@ -39,7 +39,13 @@ export async function writeTestResult(
   const { titlePath, outputDir, pageUrl } = e2eTestInfo;
   // remove the test file extensions (.spec.ts|ts, .cy.ts|js), preserving other periods in directory, file name, or test titles
   const titlePathWithoutFileExtensions = titlePath.map((pathPart) =>
-    pathPart.replace(/\.(spec|cy)\..*/, '')
+    // make sure we remove file extensions, even if the file name doesn't have .spec or .test or.cy
+    // possible extensions:
+    // playwright: https://playwright.dev/docs/test-configuration#filtering-tests
+    // cypress: https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests#Spec-files
+    pathPart
+      .replace(/\.(spec|test|cy)\..*/, '')
```

Comment:
This may not work with a title like `some.test.with.other.stuff` (extreme example, i know). But it may be safer to `.split('.')`, check the end pieces, and then join with pieces removed. 

Or a regex like this may also work, it's just getting a bit much:
`/(\.(spec|test|cy))?\.(ts|js|mjs|cjs|tsx|jsx|cjsx|coffee)$/`

Reply:
Sounds good! I'll take off the file extension first, then do a regex on the spec/test/cy part (which will then be at the end), so that'll cover things!